electroncash_gui/qt/main_window.py

```diff
@@ -2070,6 +2070,51 @@ def _chk_no_segwit_suspects(self):
             return False
         return True
 
+    def _chk_no_legacy_address(self):
+        """Show a warning if self.payto_e has legacy addresses, since the user
+        might be trying to send BTC instead of BCH."""
+        warn_legacy_address = bool(self.config.get("warn_legacy_address", True))
+        if not warn_legacy_address:
+            return
+        for line in self.payto_e.lines():
+            line = line.strip()
+            if line.lower().startswith(networks.net.CASHADDR_PREFIX + ":"):
+                line = line.split(":", 1)[1]  # strip "bitcoincash:" prefix
+            if "," in line:
+                # if address, amount line, strip address out and ignore rest
+                line = line.split(",", 1)[0]
+            line = line.strip()
+            if Address.is_legacy(line):
+                msg1 = (
+                    _("You are about to send BCH to a legacy address.")
+                    + "<br><br>"
+                    + _(
+                        "Legacy addresses are deprecated for Bitcoin Cash "
+                        "(BCH), and used by Bitcoin (BTC)."
+                    )
+                )
+                msg2 = _(
+                    "Proceed if what you are intending to do is sending BCH."
+                )
+                msg3 = _(
+                    "If you are intending to send BTC, close the "
+                    "application and use a BTC wallet instead.  Electron "
+                    "Cash is a BCH wallet, not a BTC wallet."
+                )
+                res = self.msg_box(
+                    parent=self,
+                    icon=QMessageBox.Warning,
+                    title=_("You are sending to a legacy address"),
+                    rich_text=True,
+                    text=msg1,
+                    informative_text=msg2,
+                    detail_text=msg3,
+                    checkbox_text=_("Never show this again"),
+                    checkbox_ischecked=False,
+                )
+                if res[1]:  # Never ask if checked
+                    self.config.set_key("warn_legacy_address", False)
```

Comment:
Observations:

- This will pop up a message box for EACH address it encounters that matches the predicate. This is potenrially very annoying. Loop should unconditionally break out of here after it gets a hit.
- User is being warned but with no way to stop what is about to happen. For example, if user is using a wallet with no password controls, the send will just automatically proceed.

Reply:
>     * This will pop up a message box for EACH address it encounters that matches the predicate. This is potentially very annoying. Loop should unconditionally break (or return) out of here after it gets a hit.

Thank your for catching this.  I'll fix that.

>     * User is being warned but with no way to stop what is about to happen. For example, if user is using a wallet with no password controls, the send will just automatically proceed.

Not for me when I tested manually several times.  I still get the default confirmation popup window.  I didn't want the user to have 2 confirmations windows, one after another.

>     * (Related to above point) `res[0]` is being ignored/discarded. Perhaps depending on its value, caller of this function can decide whether to abort send or not.. or something along those lines

The checkbox is for the future, for when the user tries again to use legacy address, but not for this time.  Am I missing something.

Reply:
Oh, I see you noticed and amended your comment, so my comments above are redundant.

Reply:
Ninja-edit. ;)

Reply:
Added `break` at the end of the `if` statement.